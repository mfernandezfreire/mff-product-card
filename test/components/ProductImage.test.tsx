import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';

import { product2 } from '../data/producto';



describe('ProductImage', () => {
    test('Debe mostar el componente con la imagen personalizada', () => {
        const wrapper = renderer.create(
            <ProductImage img='https://hola.jpg' className='custom-class' />
        )

        expect(wrapper.toJSON()).toMatchSnapshot();

    });

    test('Debe mostrar el componente con la imagen del producto', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2} >
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    })
})
