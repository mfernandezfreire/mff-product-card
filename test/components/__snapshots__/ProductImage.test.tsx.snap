// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe mostar el componente con la imagen personalizada 1`] = `
<img
  alt="Product"
  className="productImg custom-class"
  src="https://hola.jpg"
/>
`;

exports[`ProductImage Debe mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffe-mug.png"
  />
</div>
`;
